File: src/python/problems/largest_altitude_test.py
#https://leetcode.com/problems/find-the-highest-altitude

from core.problem_base import *

class LargestAltitude(ProblemBase):
    def Solution(self, gain: list[int]) -> int:
        result = 0
        current = 0
        for i in range(len(gain)):
            current += gain[i]
            if current > result:
                result = current

        return result
        

if __name__ == '__main__':
    TestGen(LargestAltitude) \
        .Add(lambda tc: tc.Param("gain", [-5,1,5,0,-7]).Result(1)) \
        .Add(lambda tc: tc.Param("gain", [-5,1,5,0,-7]).Result(0)) \
        .Run()

File: src/go/problems/largest_altitude_test.go
// https://leetcode.com/problems/find-the-highest-altitude/

package problems

import (
	"testing"
	"github.com/dobermanch/leetcode/core"
)

type LargestAltitude struct{}

func TestLargestAltitude(t *testing.T) {
	gen := core.TestSuite[LargestAltitude]{}
	gen.Add(func(tc *core.TestCase) {
		tc.Param([]int{-5,1,5,0,-7}).Result(1)
	}).Add(func(tc *core.TestCase) {
		tc.Param([]int{-4,-3,-2,-1,4,3,2}).Result(0)
	}).Run(t)
}

func (LargestAltitude) Solution(gain []int) int {
    result := 0
    current := 0
    for i := 0; i < len(gain); i++ {
        current += gain[i];
        if current > result {
            result = current
        }
    }

    return result
}